Makefile

```
@@ -77,18 +78,27 @@ black:
 .coverage: .coverage_pytest .coverage_filecheck
 	coverage combine --keep .coverage_pytest .coverage_filecheck
 
-# run coverage over tests
 .coverage_pytest:
+	make coverage-pytest-tests
```

```suggestion
.coverage_pytest: coverage-pytest-tests
	
```
Feels weird to call make in make

---

I agree, but this does not behave as I think it should. If I know do `make .coverage_pytest` with the .coverage_pytest file still around, it rebuilds.

---

oh sorry it should be the other way round, the target should depend on the non-phony target

---

The other way around makes it s.t. `make coverage` does not re-run the pytests for coverage. Because `.coverage_pytest` already exists.

---

that makes sense. then maybe something like this:
```
.non-phony: SAME_AS
    BEFORE

phony:
    make .non-phony
```

---

Maybe I am confused now, but this does the same thing as above, i.e. does not run pycheck tests for `make coverage`

---

It does seem a bit weird. You would usually `$(MAKE)` to recurse in Makefiles, but that’s mainly to use another Makefile than the one you currently use. Would this work?


```
# run coverage over all tests and combine data files
coverage: coverage-tests coverage-filecheck ${COVERAGE_FILE}

# merge coverage files
${COVERAGE_FILE}: ${TESTS_COVERAGE_FILE} ${FILECHECK_COVERAGE_FILE}
	coverage combine --keep --data-file=$@ ${TESTS_COVERAGE_FILE} ${FILECHECK_COVERAGE_FILE}

# run coverage over tests
coverage-tests:
	COVERAGE_FILE=${TESTS_COVERAGE_FILE} pytest -W error --cov --cov-config=.coveragerc

# run coverage over filecheck tests
coverage-filecheck:
	lit -v tests/filecheck/ -DCOVERAGE
	coverage combine --data-file=${FILECHECK_COVERAGE_FILE} .coverage.*

.PHONY: coverage coverage-tests coverage-filecheck
```

---

So making `coverage-tests` and `coverage-filecheck` re-run their part of tests, making `coverage` runs both + the merge. Anything depending on `${COVERAGE_FILE}` will re-run the merge if needed, but not the actual tests.

---

I tried to implement what you were suggesting. Note that the initial thing failed with `make: *** No rule to make target '.coverage.tests', needed by '.coverage'.  Stop.`. I added some more rules to fix this issue, but the behavior now once again is that calling `make coverage-report-md` runs the coverage target.

---

Feel free to push to this branch yourself, if you know how to make this work :)